crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1027,7 +1084,8 @@ impl<'db> Type<'db> {
             | Type::BytesLiteral(_)
             | Type::SliceLiteral(_)
             | Type::KnownInstance(_) => true,
-            Type::ClassLiteral(_) | Type::SubclassOf(_) | Type::Instance(_) => {
+            Type::SubclassOf(SubclassOfType { base }) => matches!(base, ClassBase::Class(_)),
```

Comment:
There are unit tests below for `is_fully_static` as well; let's add a couple tests (negative and positive) for this case?

Reply:
Done